lib/mastermind.rb

```diff
@@ -1,70 +1,111 @@
-	 	@colors = [ 'r' , 'o', 'y', 'g', 'b', 'p']
-	 	@remaining_guesses = 10
+	def initialize 
+		@colors = [ 'r' , 'o', 'y', 'g', 'b', 'p']
+	 	@code = []
 	end
 
 	def print_directions
-		 puts %{
-	To play: The computer will generate a four digit color code of the following colors
-	Red (r), Orange (o), Yellow (y), Green (g), Blue (b), Purple (p). You will try to
-	guess the code by writing a sequence of four lowercase letters, each representing
-	a color. Colors can be repeated and order does matter in the sequence. The computer
-	will return feedback in the form of pegs: red meaning that a color exists in the
-	secret code and is in the correct location; white meaning that the color is in the 
-	secret code but is in the wrong location. You will have ten guesses.
-
-	Additional commands:
-	/help	: to print the instructions
-	/restart: to generate a new game
-	/quit   : to quit the game\n\n}
+		message = """\nWelcome to Mastermind!\n
+		To play: The computer will generate a four digit color code of the following colors
+		Red (r), Orange (o), Yellow (y), Green (g), Blue (b), Purple (p). You will try to
+		guess the code by writing a sequence of four lowercase letters, each representing
+		a color. Colors can be repeated and order does matter in the sequence. The computer
+		will return feedback in the form of pegs: red meaning that a color exists in the
+		secret code and is in the correct location; white meaning that the color is in the 
+		secret code but is in the wrong location. You will have ten guesses.
+
+		Additional commands:
+		/help	: to print the instructions
+		/restart: to generate a new game
+		/quit   : to quit the game\n\n}"""
+
+		puts (message)
 	end
 
-	def generate_code
+end
+
+
+class SecretCode
+
+	def generate (colors)
 		secret_code = []
 		4.times do |i|
 			secret_code << colors.sample
 		end
 		secret_code
 	end
-		# CONTROLS ## MOVE TO MASTER CLASS
-		puts ("\nWelcome to Mastermind!").center(40)
-		print_directions
-		code = generate_code
+
 end
 
-Class GamePlay
-	
-	attr_reader guesses
-	attr_reader remaining_guesses
 
-	def initialize 
-		guesses = []
-		remaining_guesses = 10
+class UserGuess
+
+	attr_reader :remaining_guesses
+
+	def initialize
+		@remaining_guesses = 10
 	end
 
 	def prompt_user_guess
-		print %{Remaining guesses: #{remaining_guesses}\nPlease enter code [x x x x]: }
+		print %{Remaining guesses: #{@remaining_guesses}\nPlease enter code [x x x x]: }
 		input = STDIN.gets
-		input.split(" ")
+		input = input.split(" ")
+
+	end
+
+	def check_size(guess)
+		pass = true
+		if (guess.size != 4) 
+			puts ("INVALID INPUT: Please enter four colors")
+			pass = false
+		end
+		pass
 	end
 
-	def validate_code
-		valid 
-		guesses[-1]. eq 4
+	def check_colors(colors, guess)
+		pass = true
+		guess.each do |element| 
+			if (!(colors.include?(element)))
+				puts ("INVALID INPUT: Please enter only the first letter of the colors you wish to guess")
+				pass = false
+			end
+		end
+		pass
 	end
```

Comment:
You may not need `pass` and can instead do an explicit `return false` in the if blocks for `check_size` and `check_colors` and otherwise return true outside the if. Your tests can test the output of each method rather than the value of `pass`.